### ISSUE TITLE
Feature Request: Audit Trail for Agent Interactions

### ISSUE THREAD

**Issue by dchichkov** [steveyegge/beads#649]
Log agent interactions (prompts, responses, tool calls) to .beads/interactions.jsonl. Allow labeling entries as good or bad after the fact. This enables both auditing ("why did the agent do that?") and fine-tuning dataset generation (so models get better at Beads workflows over time).

---

### Why?

Beads tracks *what changed* (issues, code) but not *how the agent got there*. We lose the prompts, the model outputs, the tool calls, and whether they worked or failed.

An audit trail would let us:

1. **Train better models** — Real task data from real repos makes great SFT/RL training sets
2. **Debug agent behavior** — "Why did it do that?" becomes answerable
3. **Label outcomes** — Mark responses as good/bad after seeing what happened

Since Beads already syncs JSONL to git, the audit log would naturally be versioned and portable.

---

### Current State
Currently: only the LLM client stores the conversation, the prompts/responses are ephemeral and get discarded.

---

### What would it look like?

**A new file:** `.beads/interactions.jsonl`

Each line is one event. Append-only (labels are new entries, not edits).

```jsonl
{"id":"int-a1b2","kind":"llm_call","model":"claude-3-5-haiku","prompt":"Summarize this issue...","response":"**Summary:** Fixed auth bug...","issue_id":"bd-42","created_at":"2024-12-19T10:30:00Z"}
{"id":"int-c3d4","kind":"label","parent_id":"int-a1b2","label":"good","reason":"Accurate summary, merged without changes"}
```

**New CLI commands:**

```bash
# Record an interaction (or pipe JSON via stdin)
bd audit record --kind llm_call --prompt "..." --response "..."

# Label an existing entry
bd audit label int-a1b2 --label good --reason "Worked perfectly"
bd audit label int-c3d4 --label bad --reason "Hallucinated a file path"
```

**Automatic logging:** Built-in LLM features (like `bd compact`) would log their prompts/responses automatically.

---

### Auto-labeling ideas (for RL pipelines)

You don't want to manually label thousands of entries. Here are signals you can extract automatically:

| Signal | Good | Bad |
|--------|------|-----|
| Tool exit code | 0 | non-zero |
| Retries | None needed | Multiple attempts |
| PR outcome | Merged | Closed without merge |
| Reverts | None | Reverted within N days |
| Issue lifecycle | Closed, stays closed | Reopened, spawns fix issues |
| CI results | Green | Red |

A post-processing script reads `interactions.jsonl`, correlates with git history / CI, and appends label entries.

---

### Acceptance criteria

- [ ] `.beads/interactions.jsonl` exists, append-only
- [ ] `bd audit record` works
- [ ] `bd audit label` works  
- [ ] `bd sync` includes `interactions.jsonl`
- [ ] `bd compact` logs prompt/response when enabled

**Comment by steveyegge** [comment]
This feature request is being addressed by PR #651 which adds:

- `.beads/interactions.jsonl` (append-only audit log)
- `bd audit record` and `bd audit label` commands
- Integration with `bd compact --auto`

Once the PR is merged, this can be closed.
